src/widgets/ReportTableWidget.php

```diff
@@ -121,12 +125,38 @@ private function renderResultCount()
 
         echo Html::tag(
             'b',
-            $this->noOfRecords . ' record' . ($this->noOfRecords == 1 ? '' : 's') . ' returned'
+            $this->getDataCountInfo()
         );
 
         echo $this->endDiv();
     }
 
+    private function getDataCountInfo()
+    {
+        $limit = Report::PAGE_LIMIT;
+        $msgTemplate = 'Showing %s &ndash; %s of %s items';
+        $msg = sprintf($msgTemplate, 1, $this->count, $this->count);
+        if ($this->count > $limit) {
+            if ($this->page) {
```

Comment:
Can we add this condition to the one in the outer loop and avoid nesting?

Reply:
Done